SygicUIKit/Classes/Views/Search/SYUISearchResultsTableViewController.swift

```diff
@@ -0,0 +1,77 @@
+import UIKit
+
+
+/// Protocol defines generic interface for search results presenter
+public protocol SYUISearchResultsProtocol {
+    associatedtype T
+    var data: [T] { get }
+    var selectionBlock: ((T)->())? { get set }
+}
+
+
+/// Abstract class. Should be subclassed to represent search result in desired style.
+public class SYUISearchResultsViewController<Result>: UIViewController, SYUISearchResultsProtocol {
+    public var data: [Result] = []
+    public var selectionBlock: ((Result) -> ())?
+}
+
+
+/// Table view implementation of search results
+public class SYUISearchResultsTableViewController<Result: SYUIDetailCellDataSource>: SYUISearchResultsViewController<Result>, UITableViewDelegate, UITableViewDataSource {
```

Comment:
Nechces toto rozdelit do extensions nech je to trochu citatelnejsie?

Reply:
aj ked je pravda ze samotne tela tych delegatov su 1 alebo 2 funkcie takze je to asi aj zbytocne. Pisal som to cisto k tomuto riadku ale kedze su tam len 2 funkcie tak to nema moc zmysel.

Reply:
ani sa to neda pretoze extension uz ma problem pracovat s generickymi typmi